Magento_Wishlist/web/js/add-to-wishlist-mixin.js

```diff
@@ -53,9 +53,9 @@ define([
             // don't update dataToAdd if element doesn't belong to the product
             if ($(element).hasClass('super-attribute-select') && updateWishlistBtn.length) {
               var btnProduct = updateWishlistBtn.data('post').data.product;
-              var elementProduct = $(element).data('product').toString();
+              var elementProduct = $(element).data('product')?.toString();
```

Comment:
`?.` is from ES2020 so I think using it requires adding `.babel` to the filename to ensure it works in different browsers